### ISSUE TITLE
Binance: Too many requests; current limit is 1200 req/min

### ISSUE THREAD

**Issue by kascheri12** [rotkehlchenio/rotkehlchen#134]
I got the following error/warning message when creating a tax report for 8/1/2018-8/26/2018. 

# Trade History Query Warning
>During trade history query we got: Binance API request https://api.binance.com/api/v3/myTrades?symbol=PIVXBTC&fromId=0&recvWindow=10000×tamp=1535324978160&signature=de264... for myTrades failed with HTTP status code: 429, error code: -1003 and error message: Too many requests; current limit is 1200 requests per minute. Please use the websocket for live updates to avoid polling the API.. History report is probably not complete.

Should there be a setting in rotkehlchen to limit the basic requests per minute for each exchange's API? 

It seems that different tiers of membership at each exchange might allow for a greater number of available requests/min. It would be best if rotkehlchen would support faster polling, when possible.

**Comment by LefterisJP** [comment]
Where did you see this warning? Where was it shown?

 Binance does indeed have query limits as explained in their API [here](https://github.com/binance-exchange/binance-official-api-docs/blob/master/rest-api.md#limits)

**Comment by kascheri12** [comment]
This was shown as a popup in rotkehlchen.

Thank you for the API link.

**Comment by kascheri12** [comment]
Is there any way to limit the requests per minute from rotkehlchen so that the error doesn't happen?

**Comment by LefterisJP** [comment]
It keeps happening for you? What action are you taking when it happens?

**Comment by kascheri12** [comment]
Doesn't seem to happen anymore. If it happens again I'll open another issue.

**Comment by kascheri12** [comment]
I got a similar error again...
<img width="2004" alt="screen shot 2018-09-20 at 10 28 54 pm" src="https://user-images.githubusercontent.com/4165385/45915872-9fd41d00-be11-11e8-8ea1-3accfbabb286.png">

**Comment by LefterisJP** [comment]
Hmm I see what you mean. This limitation comes due to the way the binance API is written. I have to iterate through all existing markets to see which ones you have history for. Perhaps they have added a new endpoint since the time I wrote this. I will check.

**Comment by LefterisJP** [comment]
No I dont' see any such endpoint: https://github.com/binance-exchange/binance-official-api-docs/blob/master/rest-api.md

**Comment by LefterisJP** [comment]
@kascheri12 I can try to handle the 429 response for binance and wait on it instead of failing like above. But how the heck did you manage to go over 1200 requests per minute? Can you perhaps describe exactly what you did or even send me logs?

**Comment by LefterisJP** [comment]
Hey @kascheri12 I wrote a patch that should handle the 429 error: https://github.com/rotkehlchenio/rotkehlchen/pull/248

I made a mock test for it since I could not reproduce the 429 error response no matter how fast I spammed binance. Looking at your logs it seems that you can for some reason spam faster than I can.  One page of logs (~100 requests) takes 5 milliseconds for you while for me 12. I have an old computer :)

I also increased the limit of binance responses to 1000 from the default of 500. Hope they honor it. I tested with smaller limit of 5 and they do.

Unfortunately they don't honor `startTime` and `endTime` so the filtering happens on the rotkehlchen side. We query all trades that ever happened.

Thank you for testing Binance. I have only 70 trades in there and only in one pair so there may be things rotkehlchen misses. Testing with external APIs that depend on you having funds on the exchange is really hard. Wish more exchanges follow bitmex's example and offer Testing API endpoints.
